src/main/java/gov/cabinetofice/gapuserservice/web/LoginControllerV2.java

```diff
@@ -182,11 +180,14 @@ private Cookie getCustomJwtCookieFromRequest(final HttpServletRequest request) {
 
     private void verifyStateAndNonce(final String nonce, final StateCookieDto stateCookieDto, final String state) {
         // Validate that state returned is the same as the one stored in the cookie
+        final String saltId = stateCookieDto.getSaltId();
         final String encodedStateJson = oneLoginService.buildEncodedStateJson(
                 stateCookieDto.getRedirectUrl(),
-                stateCookieDto.getState()
+                stateCookieDto.getState(),
+                saltId
         );
-        final String hashedStateCookie = encryptionService.getSHA512SecurePassword(encodedStateJson);
+        final String hashedStateCookie = encryptionService.getSHA512SecurePassword(encodedStateJson, saltId);
+        encryptionService.deleteSalt(saltId);
```

Comment:
As discussed to wrap the delete in a check so we're not arbitrarily deleting salts

Reply:
done, thanks